code/src/components/ClearButton.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { todos } from 'reducers/todos'
+
+import "./clearButton.css"
+
+export const ClearButton = () => {
+
+  const dispatch = useDispatch()
+
+  const list = useSelector(state => state.todos.tasks)
```

Comment:
Since you're only using the length of the array, I'd recommend returning that directly from the selector:

```
const numberOfTasks = useSelector(store => store.tasks.items.length)
```

When you do it like you have, this `ClearButton ` component will re-render whenever the array of tasks changes (for example toggling a task's completion), but if you return the length from the selector, it'll only re-render when the length of the array changes. Small optimisation, but good to keep in the back of your head if you're working on a bigger project where you want to avoid unnecessary re-renders.

Reply:
Thank you, Damien. Wasn't sure that it was ok to do it like that. I'm making this change to this and empty list state.